nmdc_runtime/site/changesheets/scripts/missing_neon_soils_ecosystem_data.py

```diff
@@ -0,0 +1,338 @@
+
+    # Check for missing gold biosample identifiers
+    changesheet_line_items.extend(_check_gold_biosample_identifiers(goldbs, nmdcbs))
+
+    return changesheet_line_items
+
+
+def _check_ecosystem_metadata(goldbs, nmdcbs) -> List[ChangesheetLineItem]:
+    # nmdc to gold ecosystem key map
+    ecosystem_key_map = {
+        "ecosystem": "ecosystem",
+        "ecosystem_category": "ecosystemCategory",
+        "ecosystem_type": "ecosystemType",
+        "ecosystem_subtype": "ecosystemSubtype",
+    }
+    changesheet_line_items = []
+    for nmdc_key, gold_key in ecosystem_key_map.items():
+        if not gold_key in goldbs:
+            logging.warning(f"no {gold_key} for {goldbs['biosampleGoldId']}...")
+            continue
+        if not nmdc_key in nmdcbs:
+            changesheet_line_items.append(
+                ChangesheetLineItem(
+                    id=nmdcbs["id"],
+                    action="update",
+                    attribute=nmdc_key,
+                    value=goldbs.get(gold_key),
+                )
+            )
+            continue
+        if nmdcbs[nmdc_key] != goldbs.get(gold_key):
+            changesheet_line_items.append(
+                ChangesheetLineItem(
+                    id=nmdcbs["id"],
+                    action="update",
+                    attribute=nmdc_key,
+                    value=goldbs.get(gold_key),
+                )
+            )
+            continue
+
+    return changesheet_line_items
+
+
+def _check_gold_biosample_identifiers(goldbs, nmdcbs) -> List[ChangesheetLineItem]:
+    changesheet_line_items = []
+    goldbs_id = normalize_gold_id(goldbs["biosampleGoldId"])
+    if not goldbs_id in nmdcbs["gold_biosample_identifiers"]:
+        changesheet_line_items.append(
+            ChangesheetLineItem(
+                id=nmdcbs["id"],
+                action="insert",
+                attribute="gold_biosample_identifiers",
+                value=goldbs_id + "|",
+            )
+        )
+    return changesheet_line_items
+
+
+def compare_projects(gold_project, omprc_record) -> ChangesheetLineItem:
+    gold_project_id = normalize_gold_id(gold_project["projectGoldId"])
+    if "gold_sequencing_project_identifiers" not in omprc_record:
+        return ChangesheetLineItem(
+            id=omprc_record["id"],
+            action="insert",
+            attribute="gold_sequencing_project_identifiers",
+            value=gold_project_id + "|",
+        )
+
+    if gold_project_id not in omprc_record["gold_sequencing_project_identifiers"]:
+        return ChangesheetLineItem(
+            id=omprc_record["id"],
+            action="insert",
+            attribute="gold_sequencing_project_identifiers",
+            value=gold_project_id + "|",
+        )
+
+
+@click.command()
+@click.option("--study_id", default=GOLD_NEON_SOIL_STUDY_ID, help="GOLD study ID")
+@click.option(
+    "--apply_changes", is_flag=True, default=False, help=("Apply the changes")
```

Comment:
I recommend adding the "Not implemented!" caveat to the option's own `help` string. I almost missed it in the main function's docstring.